practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/service/PractitionerService.java

```diff
@@ -1,34 +1,59 @@
 package uk.ac.ox.ndph.mts.practitioner_service.service;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.UUID;
-
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.tuple.Pair;
 import org.hl7.fhir.r4.model.Enumerations.AdministrativeGender;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import uk.ac.ox.ndph.mts.practitioner_service.model.Practitioner;
 import uk.ac.ox.ndph.mts.practitioner_service.repository.FhirRepository;
-import uk.ac.ox.ndph.mts.practitioner_service.exception.ArgumentException;
+import uk.ac.ox.ndph.mts.practitioner_service.configuration.PractitionerAttribute;
+import uk.ac.ox.ndph.mts.practitioner_service.exception.ServerError;
+import uk.ac.ox.ndph.mts.practitioner_service.exception.ValidationException;
 
 /**
  * Implement an EntityService interface.
- * validation is for empty values on all fields.
+ * Validation of practitioner based on the input configuration regex fields before
+ * sending the entity to store in the fhir repository.
  */
 @Service
 public class PractitionerService implements EntityService {
+    private static final String REGEX_ALL = ".*";
     private static final String FIELD_NAME_PREFIX = "prefix";
-    private static final String FIELD_NAME_GIVEN_NAME = "given name";
-    private static final String FIELD_NAME_FAMILY_NAME = "family name";
+    private static final String FIELD_NAME_GIVEN_NAME = "givenName";
+    private static final String FIELD_NAME_FAMILY_NAME = "familyName";
     private static final String ERROR_MESSAGE = "value of argument %s cannot be empty";
+    private static final String LOG_START = "Loaded practitioner service with configuration: {}";
 
-    private final FhirRepository fhirRepository;
+    private final FhirRepository fhirRepository;    
+    private final Map<String, Pair<String, Pattern>> validationMap;
+    private final Logger logger = LoggerFactory.getLogger(PractitionerService.class);
 
     /**
      *
-     * @param fhirRepository
+     * @param fhirRepository - FHIR repository interface
+     * @param configurationProvider - provider of validation configuration
      */
     @Autowired
-    public PractitionerService(FhirRepository fhirRepository) {
+    public PractitionerService(FhirRepository fhirRepository,
+        PractitionerConfigurationProvider configurationProvider) {
```

Comment:
possibly remove business logic from constructor to keep it clean?

Reply:
this is not business logic, but initialisation logic (parsing the input config file). i prefer having the service break on init when config file is un-parseable than having to deal with strange error responses from the service after its alive.